simulator/simple_planning_simulator/test/test_simple_planning_simulator.cpp

```diff
@@ -205,183 +140,137 @@ VehicleKinematicState comToBaselink(const VehicleKinematicState & com)
-      EXPECT_EQ(pub_sub_node->pub_ackermann_command_->get_subscription_count(), expected);
-      EXPECT_EQ(pub_sub_node->pub_state_cmd_->get_subscription_count(), expected);
-      EXPECT_EQ(pub_sub_node->pub_initialpose_->get_subscription_count(), expected);
-      EXPECT_EQ(pub_sub_node->kinematic_state_sub_->get_publisher_count(), expected);
-    }
-
-    // check initial pose
-    _resetInitialpose();
-    const auto init_state = *(pub_sub_node->current_state_);
-
-    // go forward
-    _resetInitialpose();
-    _sendFwdGear();
-    _sendCommand(ackermannCmdGen(sim_node->now(), 0.0f, target_vel, target_acc));
-    isOnForward(*(pub_sub_node->current_state_), init_state);
-
-    // go backward
-    _resetInitialpose();
-    _sendBwdGear();
-    _sendCommand(ackermannCmdGen(sim_node->now(), 0.0f, -target_vel, -target_acc));
-    isOnBackward(*(pub_sub_node->current_state_), init_state);
-
-    // go forward left
-    _resetInitialpose();
-    _sendFwdGear();
-    _sendCommand(ackermannCmdGen(sim_node->now(), target_steer, target_vel, target_acc));
-    isOnForwardLeft(*(pub_sub_node->current_state_), init_state);
-
-    // go backward right
-    _resetInitialpose();
-    _sendBwdGear();
-    _sendCommand(ackermannCmdGen(sim_node->now(), -target_steer, -target_vel, -target_acc));
-    isOnBackwardRight(*(pub_sub_node->current_state_), init_state);
+  const auto vehicle_model_type = GetParam();
+
+  std::cout << "\n\n vehicle model = " << vehicle_model_type << std::endl << std::endl;
+  rclcpp::NodeOptions node_options;
+  node_options.append_parameter_override("initialize_source", "INITIAL_POSE_TOPIC");
+  node_options.append_parameter_override("vehicle_model_type", vehicle_model_type);
+  node_options.append_parameter_override("initial_engage_state", true);
+  node_options.append_parameter_override("add_measurement_noise", false);
+  declareVehicleInfoParams(node_options);
+  const auto sim_node = std::make_shared<SimplePlanningSimulator>(node_options);
+
+  const auto pub_sub_node = std::make_shared<PubSubNode>();
+
+  const float32_t target_vel = 5.0f;
+  const float32_t target_acc = 5.0f;
+  const float32_t target_steer = 0.2f;
+
+  auto _resetInitialpose = [&]() { resetInitialpose(sim_node, pub_sub_node); };
+  auto _sendFwdGear = [&]() { sendGear(GearCommand::DRIVE, sim_node, pub_sub_node); };
+  auto _sendBwdGear = [&]() { sendGear(GearCommand::REVERSE, sim_node, pub_sub_node); };
+  auto _sendCommand = [&](const auto & _cmd) { sendCommand(_cmd, sim_node, pub_sub_node); };
+
+  // check pub-sub connections
+  {
+    size_t expected = 1;
+    EXPECT_EQ(pub_sub_node->pub_ackermann_command_->get_subscription_count(), expected);
+    EXPECT_EQ(pub_sub_node->pub_gear_cmd_->get_subscription_count(), expected);
+    EXPECT_EQ(pub_sub_node->pub_initialpose_->get_subscription_count(), expected);
+    EXPECT_EQ(pub_sub_node->current_odom_sub_->get_publisher_count(), expected);
```

Comment:
FYI: 
I am proposing a framework to check the pub-sub connections.
If you are interested, please read the design document.
https://github.com/autowarefoundation/autoware.universe/pull/391